app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -12,45 +12,119 @@ import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.lazy.GridCells
 import androidx.compose.foundation.lazy.LazyVerticalGrid
 import androidx.compose.foundation.lazy.items
 import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Snackbar
+import androidx.compose.material.SnackbarDuration
+import androidx.compose.material.SnackbarHost
 import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.material.rememberScaffoldState
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import kotlinx.coroutines.launch
 import org.mozilla.fenix.R
 import org.mozilla.fenix.theme.FirefoxTheme
 import org.mozilla.fenix.wallpapers.Wallpaper
 
 /**
- * The screen for controlling settings around Wallpapers.
+ * The screen for controlling settings around Wallpapers. When a new wallpaper is selected,
+ * a snackbar will be displayed.
  *
  * @param wallpapers Wallpapers to add to grid.
  * @param selectedWallpaper The currently selected wallpaper.
- * @param onSelectWallpaper Action to take when a new wallpaper is selected.
+ * @param onSelectWallpaper Callback for when a new wallpaper is selected.
+ * @param onViewWallpaper Callback for when the view action is clicked from snackbar.
  */
 @Composable
 fun WallpaperSettings(
     wallpapers: List<Wallpaper>,
     selectedWallpaper: Wallpaper,
     onSelectWallpaper: (Wallpaper) -> Unit,
+    onViewWallpaper: () -> Unit,
 ) {
-    Surface(color = FirefoxTheme.colors.layer2) {
-        WallpaperThumbnails(
-            wallpapers = wallpapers,
-            selectedWallpaper = selectedWallpaper,
-            onSelectWallpaper = onSelectWallpaper
-        )
+    val coroutineScope = rememberCoroutineScope()
+    val scaffoldState = rememberScaffoldState()
+
+    Scaffold(
+        backgroundColor = FirefoxTheme.colors.layer1,
+        scaffoldState = scaffoldState,
+        snackbarHost = { hostState ->
+            SnackbarHost(hostState = hostState) {
+                WallpaperSnackbar(onViewWallpaper)
+            }
+        }
+    ) {
+        Surface(color = FirefoxTheme.colors.layer2) {
+            WallpaperThumbnails(
+                wallpapers = wallpapers,
+                selectedWallpaper = selectedWallpaper,
```

Comment:
This should be consider a followup or future consideration, but one way to avoid passing the `sellectedWallpaper` would be to have `isSelected` state in `Wallpaper`.